tls/extensions/s2n_cert_status.c

```diff
@@ -40,7 +42,8 @@ const s2n_extension_type s2n_cert_status_extension = {
 
 static bool s2n_cert_status_should_send(struct s2n_connection *conn)
 {
-    return s2n_server_can_send_ocsp(conn);
+    return conn->handshake_params.our_chain_and_key
+            && conn->handshake_params.our_chain_and_key->ocsp_status.size > 0;
```

Comment:
I feel like I'm missing something obvious here. Wouldn't this always send the ocsp info if available, whether or not it was requested, because you're no longer checking status_request? You mention below "A server can request OCSP stapling from a client, but it is not tracked with this variable". How is it tracked?

Do we have a test where ocsp info is available but not requested?

Reply:
I'm relying on the `is_response` extension variable to determine whether to send it or not. If the user supplied an OCSP response, it will always send a response upon a request. But without a request, the extension will never be sent.

A test for this definitely makes sense. I added a new test that sets a response on the server but doesn't request a response on the client, and then ensures the response wasn't set.

Reply:
I knew I was missing something obvious :) That makes sense.